### ISSUE TITLE
[BUG] [v2.32.1] run service that extends other service does not fulfill depends_on

### ISSUE THREAD

**Issue by buxm** [docker/compose#12423]
### Description

With version 2.32.1, when running `docker compose run <serviceA>` and `<serviceA>` extends `<serviceA_base>` and `<serviceA_base>`  `depends_on` `serviceB`, the `depends_on` clause is not fulfilled.
I.e. `serviceA` runs without starting `serviceB`.
This did not happen with prior versions (2.31.0 and before).

### Steps To Reproduce

1. On Linux (I tested with Ubuntu 24.04.1) install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin. docker-compose-plugin must be version 2.32.1
2. put in the `test` directory the following `compose.yaml` file:
```
### compose.yaml
services:
  redis:
    image: redis

  job-base:
    image: busybox
    depends_on:
      redis:
        condition: service_started
    restart: no

  hello-world-job:
    extends:
      service: job-base
    command: echo Hello world
```
Because `hello-world-job` extends `job-base`, `hello-world-job` depends on the service `redis`, that must have been started before starting `hello-world-job`.
3. from within the `test` directory, run `docker compose run --rm hello-world-job`. The output is:
```
[+] Creating 1/1
 ✔ Network test_default  Created                                                                                                                                    0.2s
Hello world
```
`hello-world-job` has run but the container for the service `redis` has not been created.
4. install docker-compose-plugin version 2.31.0 (`sudo apt install docker-compose-plugin=2.31.0*`)
5. run `docker compose run --rm hello-world-job`. The output is:
 ```
[+] Creating 2/2
 ✔ Network test_default    Created                                                                                                                                  0.2s
 ✔ Container test-redis-1  Created                                                                                                                                  0.1s
[+] Running 1/1
 ✔ Container test-redis-1  Started                                                                                                                                  0.3s
Hello world
```
The container for the service `redis` has been created and started before running `hello-world-job`. This is the expected result.

### Compose Version

```Text
Docker Compose version v2.32.1
```

### Docker Environment

```Text
Client: Docker Engine - Community
 Version:    27.4.1
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.19.3
    Path:     /usr/libexec/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.32.1
    Path:     /usr/libexec/docker/cli-plugins/docker-compose

Server:
 Containers: 10
  Running: 6
  Paused: 0
  Stopped: 4
 Images: 11
 Server Version: 27.4.1
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: journald
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 88bf19b2105c8b17560993bee28a01ddc2f97182
 runc version: v1.2.2-0-g7cb3632
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: builtin
  cgroupns
 Kernel Version: 6.8.0-51-generic
 Operating System: Ubuntu 24.04.1 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 2
 Total Memory: 3.73GiB
 Name: e4wardtest01
 ID: 1722a4b8-f46f-4d00-8036-52be6254174b
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: true
```

**Comment by paulbehrisch** [comment]
i'm getting the same behavior. Everything worked with Compose 2.31.0 and with 2.32.1 services are not being started.

**Comment by klutchell** [comment]
I came here to open the exact same issue, thanks for opening this!

Here is my minimal reproduction:
```yaml
# docker-compose.a.yml
services:
  service1:
    image: nginx:alpine

```
```yaml
# docker-compose.b.yml
services:
  service2:
    image: nginx:alpine
  service3:
    extends:
      file: docker-compose.a.yml
      service: service1
    depends_on:
      - service1
```
#### With v2.31.0 the `depends_on` property is present.
```
Docker Compose version v2.31.0
name: src
services:
  service1:
    image: nginx:alpine
    networks:
      default: null
  service2:
    image: nginx:alpine
    networks:
      default: null
  service3:
    depends_on:
      service1:
        condition: service_started
        required: true
    image: nginx:alpine
    networks:
      default: null
networks:
  default:
    name: src_default
```
#### With v2.32.1 the `depends_on` property is not included.
```
Docker Compose version v2.32.1
name: src
services:
  service1:
    image: nginx:alpine
    networks:
      default: null
  service2:
    image: nginx:alpine
    networks:
      default: null
  service3:
    image: nginx:alpine
    networks:
      default: null
networks:
  default:
    name: src_default
```

**Comment by thaJeztah** [comment]
- Looks possibly related to / duplicate of https://github.com/docker/compose/issues/12421

**Comment by buxm** [comment]
Yes, I think it's duplicate of #12421, although the `depends_on` is placed differently.
Sorry, I failed to check closed issues before opening a new one.

**Comment by buxm** [comment]
My understanding is that v2.32.3 should have fixed this issue. However v2.32.4 still shows the same behaviour.

**Comment by paul-mcculloch** [comment]
I can also confirm this bug is present in 2.32.3 & 2.32.4  (Ubuntu 2.32.4-1~ubuntu.24.04~noble)

**Comment by ndeloof** [comment]
tested with latest release, and confirmed to be fixed:
```
$ cat compose.yaml 
services:
  redis:
    image: redis

  job-base:
    image: busybox
    depends_on:
      redis:
        condition: service_started

  hello-world-job:
    extends:
      service: job-base

$ docker compose config
services:
  hello-world-job:
    depends_on:
      redis:
        condition: service_started
        required: true
    image: busybox
...
```

I'm closing this issue as "fixed" - feel free to open a new issue with a reproduction example if you still don't get the expected behavior
